packages/vuetify/src/components/VDataTable/__tests__/VDataTable.spec.ts

```diff
@@ -345,4 +345,74 @@ describe('VDataTable.ts', () => {
 
     expect(fn).toHaveBeenCalled()
   })
+
+  it('should not search column with filterable set to false', async () => {
+    const wrapper = mountFunction({
+      propsData: {
+        items: testItems,
+        headers: [
+          {
+            text: 'Dessert (100g serving)',
+            align: 'left',
+            filterable: false,
+            value: 'name',
+          },
+          { text: 'Calories', value: 'calories' },
+          { text: 'Fat (g)', value: 'fat' },
+          { text: 'Carbs (g)', value: 'carbs' },
+          { text: 'Protein (g)', value: 'protein' },
+          { text: 'Iron (%)', value: 'iron' },
+        ],
+      },
+    })
+
+    expect(wrapper.html()).toMatchSnapshot()
+
+    wrapper.setProps({
+      search: 'cup',
+    })
+    await wrapper.vm.$nextTick()
+
+    expect(wrapper.html()).toMatchSnapshot()
+  })
+
+  it('should not search column with filterable set to false and has filter function', async () => {
```

Comment:
this one hasn't got actual search in it, could you explain what're you testing here. Maybe I mis-understood smth here.